qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+        self.ccd.atmcd64d.start_acquisition()
+
+        # wait for single acquisition
+        if acquisition_mode == 'single scan':
+            self.ccd.atmcd64d.wait_for_acquisition()
+
+        # wait for accumulate acquisition
+        elif acquisition_mode == 'accumulate':
+            number_accumulations = self.ccd.number_accumulations.get()
+            for i in range(number_accumulations):
+                self.ccd.atmcd64d.wait_for_acquisition()
+
+        # get and return spectrum
+        return self.ccd.atmcd64d.get_acquired_data(self.ccd.x_pixels)
+
+
+class Andor_DU401(Instrument):
+    """
+    Instrument driver for the Andor DU401 BU2 CCD.
+
+    Args:
+        name (str): Instrument name.
+        dll_path (str): Path to the atmcd64.dll file. If not set, a default path is used.
+        camera_id (int): ID for the desired CCD.
+        setup (bool): Flag for the setup of the CCD. If true, some default settings will be sent to the CCD.
+
+    Attributes:
+        serial_number (int): Serial number of the CCD.
+        head_model (str): Head model of the CCD.
+        firmware_version (int): Firmware version of the CCD.
+        firmware_build (int): Firmware build of the CCD.
+        x_pixels (int): Number of pixels on the x axis.
+        y_pixels (int): Number of pixels on the y axis.
+
+    """
+
+    # TODO (SvenBo90): implement further acquisition modes
+    # TODO (SvenBo90): implement further read modes
+    # TODO (SvenBo90): implement further trigger modes
+    # TODO (SvenBo90): add and delete parameters dynamically when switching acquisition mode or read mode
+    # TODO (SvenBo90): handle shutter closing and opening timings
+
+    def __init__(self, name, dll_path=None, camera_id=0, setup=True, **kwargs):
+
+        super().__init__(name, **kwargs)
+
+        # link to dll
+        self.atmcd64d = atmcd64d(dll_path=dll_path)
+
+        # initialization
+        self.atmcd64d.initialize(' ')
+        self.atmcd64d.set_current_camera(self.atmcd64d.get_camera_handle(camera_id))
+
+        # get camera information
+        self.serial_number = self.atmcd64d.get_camera_serial_number()
+        self.head_model = self.atmcd64d.get_head_model()
+        self.firmware_version = self.atmcd64d.get_hardware_version()[4]
+        self.firmware_build = self.atmcd64d.get_hardware_version()[5]
+        self.x_pixels, self.y_pixels = self.atmcd64d.get_detector()
+
+        # add the instrument parameters
+        def accumulation_cycle_time_parser(ans):
+            return float(ans[1])
+        self.add_parameter('accumulation_cycle_time',
+                           get_cmd=self.atmcd64d.get_acquisition_timings,
+                           set_cmd=self.atmcd64d.set_accumulation_cycle_time,
+                           get_parser=accumulation_cycle_time_parser,
+                           unit='s',
+                           label='accumulation cycle time')
+
+        self.add_parameter('acquisition_mode',
```

Comment:
isn't it possible to read form the instrument the current value of this parameter?